addons/stock/models/stock_move_line.py

```diff
@@ -425,12 +427,18 @@ def _action_done(self):
                         # tracked products without a `lot_id`.
                         continue
 
-                    if not ml.lot_id:
-                        raise UserError(_('You need to supply a Lot/Serial number for product %s.') % ml.product_id.display_name)
+                    if not ml.lot_id and (ml not in ml_to_create_lot or not ml.lot_name):
+                        ml_no_lot |= ml
             elif qty_done_float_compared < 0:
                 raise UserError(_('No negative quantities allowed'))
             else:
                 ml_to_delete |= ml
+
+        if ml_no_lot:
+            raise UserError(_('You need to supply a Lot/Serial number for product: \n - ' +
+                              '\n - '.join(ml_no_lot.mapped('product_id').mapped('display_name'))))
```

Comment:
you can chain mapped .mapped('product_id.display_name')

Reply:
Aha, Thanks

Reply:
Change applied.